server/routes/v1/version.spec.ts

```diff
@@ -1,22 +1,46 @@
+import '../../utils/mockMongo'
+
 import mongoose from 'mongoose'
+
+import ModelModel from '../../models/Model'
 import UserModel from '../../models/User'
 import VersionModel from '../../models/Version'
 import * as approvalService from '../../services/approval'
-import '../../utils/mockMongo'
-import { testUser, testManager, testReviewer, testVersion } from '../../utils/test/testModels'
+import * as deploymentService from '../../services/deployment'
+import * as modelService from '../../services/model'
+import { testManager, testModel, testReviewer, testUser, testVersion } from '../../utils/test/testModels'
 import {
+  authenticatedDeleteRequest,
   authenticatedGetRequest,
   authenticatedPostRequest,
   authenticatedPutRequest,
   validateTestRequest,
 } from '../../utils/test/testUtils'
 
+jest.mock('../../services/model', () => ({
+  removeVersionFromModel: jest.fn(() => Promise.resolve()),
+  serializedModelFields: () => ({
+    mandatory: ['_id', 'uuid', 'latestVersion.metadata.highLevelDetails.name', 'schemaRef'],
+  }),
+}))
+
+jest.mock('../../services/deployment', () => ({
+  findDeploymentsByModel: jest.fn(() => Promise.resolve({ irrelevant: 'content' })),
```

Comment:
You can just make this a jest function for now as you're setting the return values in your tests (which is the right way to do it anyway) so these values will be overwritten at that point. `findDeploymentsByModel: jest.fn()`

The same applies to the other functions you're mocking in this file. We only tend to set a specific return value at this level is we want the same return value for all tests in this file. Otherwise just setting `jest.fn()` here and using `mockReturnValueOnce` or `mockImplementationOnce` in each test that uses them is the preferred approach.

Reply:
After discussion, decided to leave as is to avoid repeated code in multiple tests